TLM/TMPE.API/Traffic/Enums/JunctionRestrictionFlags.cs

```diff
@@ -0,0 +1,10 @@
+namespace TrafficManager.API.Traffic.Enums {
+    public enum JunctionRestrictionFlags {
+        AllowUTurn = 1 << 0,
+        AllowNearTurnOnRed = 1 << 1,
+        AllowFarTurnOnRed = 1 << 2,
+        AllowForwardLaneChange = 1 << 3,
+        AllowEnterWhenBlocked = 1 << 4,
+        AllowPedestrianCrossing = 1 << 5,
+    }
```

Comment:
I need to control TrafficLights too. not sure if it would fit here. segmnetId would be irrelevant.

Reply:
maybe this could be a rule override hook?

Reply:
> maybe this could be a rule override hook?

I anticipate that in general, hooks would correspond to managers. There might be exceptions, but since that's the existing organization of the API, it's the pattern that makes sense.

Reply:
I am patching Traffic lights too. and could be causing problems because my patch will break if we overload those methods. I can easily fix my patch but I don't know about NCR.

Traffic light manager does not seem to manage default value for traffic lights though. so that is a difference. I am only patching if traffic lights are configurable.